### DISCUSSION TITLE
Org and Project name - When/Should we change?

### DISCUSSION THREAD

**Discussion by chri2** [tinyBlogPub/microblog.pub#10] [General]
I'm thinking about whether the name of the fork should be different as long as Thomas didn't officially give up on the old project to have some distinction between the two of them.

**Comment by nigini** [comment]
I have thought about that a lot. But I realized that I don't want (at least for now) to start a new project; I want to contribute to this one. What I tried to do was to put everywhere that we are keeping this alive in the best interest. I even asked for his blessing, but no answers. Also, consider: Wouldn't it be strange(st) to copy some other people's software, rebrand without changes, and act as if it is our project?

If Thomas ever gets back and wants us out, that would be a good point in time to rethink our role and our branding. Or, in the near future, we will realize that we are actually ready for a new name, and I'd be fine with that.  (Deep down inside, I still hope to create something good here, and that Thomas will either hand over the name or join us! :P ).

Finally, remember: we are just using this "central" repo to facilitate joining forces to make improvements in open-source software that we all use.

PS.: @chri2 I want you to know I respect your caution, and I hope you see that I am also putting a lot of thought into this. If more people think that changing the name now is a better approach, I can be convinced. I hope that will not deter you from working together because of that.

**Reply by 3b0b** [reply]
I agree entirely with...all of this; with the addition that I also recognize (and am fine with) the possibility that Thomas will find time for the project again, ask us to change this org name or dissolve completely, and want patches emailed upstream to sourcehut for anything we'd like him to consider incorporating of our efforts.

My opinion is that _until and unless_ either Thomas comes back and resumes development in a direction different from ours, _or_ we feel like changes we've made have reached the level of major or philosophical differences, it is at least okay and possibly preferable for this to continue as-is. Maybe even with some of what was clarified in the above comment added to the org description.

**Comment by nigini** [comment]
I just wanna say that it makes me really satisfied to return to these conversations. Makes me feel that we are already creating our directions for this project.

Also, I'd not mind at all changing the name of the ORG quicker, but **I'd wait to see what would be a nice name that represents our group's interest.** For example, I am highly interested in the DWEB movement and got into MicroBlog and MicroPub because I believe the Fediverse is a great way to decentralize and reclaim the space we game to big-platforms. I can envision many other projects under that umbrella, but I am sure this is not necessarily the energy that everyone here is following. So, again, let's find that path that we all agree upon.

Speaking of which, do you want to share **what got you into MicroPub? What other things would you like to contribute to?**

**Reply by 3b0b** [reply]
Personally, I make almost no use of micropub; I almost exclusively post from the admin interface. I won't go into all of the reasons I blog/microblog here, because that's what the first article on my site is (mostly) about. But I will share the reason I use `microblog.pub` here because that much is relatively simple: I wanted a blog that I could have be my Fediverse presence and also interact with Indieweb sites via webmentions, ideally self-hosted and without relying much (if at all) on other services. This was almost a perfect fit.

There is a little bit of friction from the fact that it is sort of Fediverse first and Indieweb second, because it means if I want things to thread properly for the Fediverse audience, when I reply to someone else's blog post that is Indieweb first and federated second, I _sort of_ have to elevate their Fediverse identity (or at least the Fediverse representation of their post) over their Indieweb domain, which is not necessarily ideal from some of their perspectives.

**Reply by nigini** [reply]
I am not super familiar with the Indieweb/webmention standard, but from what I've seen, it should stay as a priority. @3b0b I think this "friction" you have raised here deserves a separate Discussion post. If there is anything we can do to improve that, we should. Do you want to start it?

BTW: loved [your post](https://bw3.dev/articles/2D9hz/state-of-the-blog-address) mentioning our effort.

**Reply by 1dom** [reply]
@3b0b I liked your post. I think these 2 PRs might be relevant:

#20 - I think this might be fixing the custom route support you mentioned broke.
#22 - switches the footer to this version of code.

**Comment by nigini** [comment]
So... now that some of us have demonstrated interest in working on open web standards (mostly me and @1dom - see #16 and #14), would anyone (@chri2 @3b0b) be opposed to changing the org name to something more accommodating of that? 

Ideas: OpenBlog.pub, openWEblog, OpenWebPub?

**Reply by 1dom** [reply]
I think we should definitely rename. Assuming tsileo doesn't return, we'll be developing software called "microblog.pub" that's similar but different to the software hosted at the domain microblog.pub.  We should do it sooner rather than later too so we can all stop having to keep updating remotes :sweat_smile:

My name idea: OpenSelfPost 

I feel "post" better captures it being used for a variety of content. I'm also big on self-hosting stuff, so "OpenSelfPost" feels like a play on words. Also considered OpenIndiePost, but it doesn't sound as nice.

I feel like there's more to microblog.pub than blogging, so I like OpenWebPub the most out of your ideas so far.

**Reply by 3b0b** [reply]
I am inclined to agree that at the point we start broadening the targeted scope to other protocols, and overhauling the architecture to that end, it's probably time to consider the new name. I recommend trying to come up with something "searchable" and trying to avoid collision with standards or other technologies - the current name is not exactly difficult to distinguish from micro.blog, but it has sometimes been necessary to clarify that no it wasn't a typo. And then there's the mild potential for confusion with micropub or microblogging in general. I kinda like OpenSelfPost but my weird brain keeps trying to come up with a "ShelfPost" joke, or going off on tangents about "protoblog" (which is already at least one thing if not more) or various other ideas. I should probably refrain from contributing suggestions as I am notoriously bad at naming things.

**Comment by nigini** [comment]
My suggestion is to **first change the ORG name**, as the project is branched and still the same as the original mbp!

I don't thing the *open* term helps a lot... it makes more sense as *OpenWeb*... In this direction, there are *SocialWeb* and *DWeb*, but they are probably too big and hard to differentiate. (Although clearly developing as part of a movement is helpful for visibility.) What I like about having the term *WEB* is that it encompasses the act of connecting with the rest of the... well... web!
 
I like the self-hosting as well, although I think We-Hosting (as in, smaller groups, ex.: small Fediverse server) is a more practical idea... What about using some term in the direction of *home* or *tiny* or even *your* instead of *self*?

What about:
- YourWeb... WeWeb... (tech?) there are businesses in these spaces
-  tinyweb!!!! (I'm excited about this one! :P

**Comment by chri2** [comment]
No matter which direction the project takes, it would help to give it a different name:

I upgraded the yunhost package for microblog.pub to version 2 and will try to keep up with the yunohost development to keep running my microblog.pub on my yunhost.

Same probably applies to any other packaging for a distribution.

If there's a branch it might make sense to package that branch separately to give the choice to any user which branch to use. Packaging the another flavor of the same software origin under the same name leads to problems.

A name change is a very practical thing as long as people are running microblog.pub .

I'm using microblog.pub as a blog federated via ActivityPub and do not have much interaction over Webmention. But I'm using `micropub-go` to publish sometimes.

microblog.pub → tinyblog.pub ?

**Reply by nigini** [reply]
It would be great to have you supporting the Yunhost. I can create the branch and give you admin powers for that! Would that help?

(BTW: @chri2 @1dom @3b0b I don't feel great about being the sole king of this kingdom. I'd like, though, to have a call and know those who want to get superpowers better. Ping me if you want!)

To make it clear, I am suggesting changing the ORG name first, as I still think we are mostly all working on and running the original `microblog.pub` software. That is why I suggested `tinyweb` (a larger thing than the blog). Then, I think when we feel like we moved with the `microblog.pub` far enough from its origin, `tinyblog` sounds like a perfect name.

What do you think?

**Reply by chri2** [reply]
To be honest - I do not care much about the name.

What I like about **tinyblog.pub** is that it strongly emphasizes its heritage. I always liked on _microblog.pub_ the _.pub_ part.

Also: the domain tinyblog.pub is still free to host information about the software ;-) .

Anyway - You're putting all the work and time in it so you should choose freely.

And yes, I plan to support it on its own yunohost package, but right now I can't get into it. But I will.

Since you decided to use github (I still would strongly recommend codeberg.org or - if geeky is liked - sr.ht) the repository for integration into yunohost can live in the yunohost-app organisation. Officially supported packaged have to be put there anyway to get on the app catalog of yunhost.

For [flohmarkt](https://apps.yunohost.org/app/flohmarkt) I develop on codeberg.org inside the [same organization like the source of the project](https://codeberg.org/flohmarkt).

For microblog.pub I do the same with a repository on https://sr.ht, but there's no official relationship to Thomas and his repository (beside that I started off with the integration from his work on yunhost package format version 1).

If you'd want to start an organization for your branch of microblog.pub anyway we can put a repository containing the development of the integration beside it (like maybe sometimes the files for a debian package or a docker version or whatever integration/packaging people would like to add).

**Reply by 3b0b** [reply]
As I said before, naming things is hard. And unfortunately, I'm temperamentally more likely to be able to caution against potential problems with some names (which, to be clear, may be minor enough not to make them bad choices) than to suggest better ones. The kinds of things my brain comes up with are puns and things like "Socious" (not sure how I came up with that one) or "Sociolo" (a portmanteau of Social and Solo) or "Picroblob.mug."

The problem with going for a nice simple name that is based on relevant terms like "open" and "social" and "web" and "tiny" and "small" and "blog" is name collision.

`Microblog.pub` has going for it, despite the software also being good for long-form articles, the fact that "microblogging" is a common term and "microblogpub" made it distinctive. `Tinyblog.pub` got me thinking about "wee-blog" as a play on both "wee" as in small and "we" as in first person plural, but I don't think it would really be a good idea. `tinyblog.pub` - and referring to it as TinyBlogPub, not TinyBlog, since I believe that's already at least three other projects and at least one or two specific blogs - might be my favorite idea so far. I wouldn't say I'm enthusiastic about it, but it seems reasonable and I agree with the positives already mentioned.

**Reply by 1dom** [reply]
I like tinyblog for the relation to microblog. Whilst we're throwing nerdy puns, in that direction, another option is ublog, u as in μ/mu/micro. But the pronunciation would end up being the same as microblog :nerd_face: :facepalm:

@chri2 if it's any consolation, I recently switched my internal stuff from gitea to forgejo. I push a lot more frequently to an internal forgejo instance than I ever do to github. IMO, the only reason to use github is for work, discoverability, and collaboration with people who aren't that fussed about certain principals. Practically, codeberg, gitlab, and github all feel pretty similar. Sourcehut feels practically quite different though - not necessarily bad.

Honestly, I don't have a strong opinion about what name, but I guess it does make sense to get it all changed as it feels like this is probably going to go in a slightly different direction to microblog.pub. Even if it's just being primarily hosted on github rather than a more open place, I'm sure @chri2 isn't the only microblog.pub user with a strong preference to avoid Github.

**Reply by 1dom** [reply]
Thought about this a bit more over the weekend. Let's go tinyblog.pub, shortened to tinyblogpub or tbp, to avoid being confused with various other "tinyblog" projects.

Org name could be a shortened version of tinyblog.pub, but obviously it must be available. 

On one hand, repo name could stay as microblog-pub whilst changes are largely tweaks/improvements/fixes to original microblog.pub. But on the other hand, the sooner we change the name of the repo/package, the sooner it avoids any problems with existing stuff.

If we do a complete rewrite or similar as discussed in other threads, if not already, those efforts should definitely be in  a new repository (name tinyblogpub or similar). 

To avoid discussing this any further for a while, easiest choice is for org and repo to be renamed tinyblogpub, tinyblog, tbp or some variant - the sooner the better:)

Personally, I like thinking about how this could all be rearchitecting a rewritten front scratch, but practically, for the next probably 6 months or so at least, I'll just be doing tweaks & fixes to the existing code base, getting a few staging instances deployed and making sure I actually understand the existing different features and functionality.

**Comment by nigini** [comment]
Thank you for your thoughtful input: **tinyblog.pub**, tinyBlogPub, tBP it is! Next steps:

- will change the ORG name here,
- will point the domain to this ORG,
- keep the current repo for maintenance of mBP
- create a new repo(s) as we make decisions about TBP ( rule 1: *(hope you agree:) only move-add code that is fully tested!* ;)

I like to have the GitHub home mostly for visibility, but I am open to moving the work to a less-big-corp environment in the near future.

I am very excited about the next steps!!!
